File: day01.py
import sys


def parse_input(lines):
    elves = []
    current_elf = []
    for line in lines:
        line = line.strip()
        if not line:
            elves.append(current_elf)
            current_elf = []
        else:
            current_elf.append(int(line))
    elves.append(current_elf)
    return elves


def most_calories(elves):
    return max(sum(elf) for elf in elves)


def top_three(elves):
    sorted_elves = sorted((sum(elf) for elf in elves), reverse=True)
    return sum(sorted_elves[:3])


if __name__ == '__main__':
    elves = parse_input(sys.stdin)
    print(most_calories(elves))
    print(top_three(elves))

File: day01.rs
use std::io;

fn main() {
    let mut elves = parse_input().unwrap();
    elves.sort_unstable_by(|a, b| b.cmp(a));
    println!("{}", elves[0]);
    println!("{}", elves[0..3].iter().sum::<i32>());
}

fn parse_input() -> Result<Vec<i32>, io::Error> {
    let lines = io::stdin().lines();

    let mut elves = vec![];
    let mut current_elf = 0;
    for line in lines {
        let line = line?;
        let line = line.trim();
        if line.is_empty() {
            elves.push(current_elf);
            current_elf = 0;
        } else {
            current_elf += line.parse::<i32>().unwrap();
        }
    }
    elves.push(current_elf);
    Ok(elves)
}
